packages/e2e-test-utils/src/post-type.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Internal dependencies
+ */
+import { wpDataSelect } from './wp-data-select';
+
+/**
+ * Returns a promise which resolves with the current post type name.
+ *
+ * @return {Promise<?string>} Promise resolving with current post type name.
+ */
+export async function getCurrentPostType() {
+	return wpDataSelect( 'core/editor', 'getCurrentPostType' );
+}
+
+/**
+ * Returns a promise which resolves with the post type object for the given post type name.
+ *
+ * @param  {string}           name Name of the post type.
+ * @return {Promise<?Object>}      Promise resolving with post type object.
+ */
+export async function getPostType( name ) {
+	return wpDataSelect( 'core', 'getPostType', name );
+}
```

Comment:
There are two important things here:

* In E2E tests we strongly discourage tapping into the data layer for reads and effects. This is mainly because the point of E2E tests is to simulate the experience of users, and so it should be possible to trigger change and observe results via the editor UI — even if that feels more fragile to us as developers, it ensures more faithful tests.

* Even if/when it seems unavoidable that a given test would benefit from such a function, we need to be careful about adding helpers to `e2e-test-utils`, because from that point on they become public interfaces that we will have to maintain.

Specifically, in the context of the tests of this PR, I would say:

* We don't need to read the post-type labels, and so we don't need these selector-like helpers.
* Even if we did, such logic would have to live in the file of the respective test suite.